core/src/token_info/cached.rs

```diff
@@ -37,14 +37,20 @@ impl TokenInfoCache {
         }
     }
 
+    pub async fn cache(&self) -> HashMap<TokenId, TokenBaseInfo> {
+        self.cache.read().await.clone()
+    }
+
     /// Attempt to retrieve and cache all token info that is not already cached.
     /// Fails if `all_ids` fails. Does not fail if individual token infos fail.
-    pub async fn cache_all(&self, number_of_parallel_requests: usize) -> Result<()> {
-        stream::iter(self.all_ids().await.context("failed to get all ids")?)
+    /// On success returns the result of the internal `all_ids` call.
+    pub async fn cache_all(&self, number_of_parallel_requests: usize) -> Result<Vec<TokenId>> {
```

Comment:
As you can see below, we need the full cache and all_ids so I think it's a good idea to reuse this data when we already have it for faster startup and less stress test and on the node.

Reply:
I'm not sure I understand the suggestion. I'd not be  a fan of making `cache` part of the `TokenInfoFetching` trait sine that would be leaking behavior of one specific implementation to the general abstraction. 

Which data do you want to reuse - the call to `num_tokens`?

Reply:
I mean a function that would return `HashMap<TokenId, TokenBaseInfo>` for all token ids, instead of just a single TokenBaseInfo. This function could have a default implementation in the trait and be overriden in the cache.
We reuse both all_token_ids and the full map.

Reply:
Good idea, this way we can also use batched calls in the onchain implementation once ethcontracts has better support for it.
I think there are some use cases (e.g. getting info for all tokens in the current orderbook), where we don't want all but only a subset of tokens.
How about making `get_token_info` plural and have it take a vector and return a vec or hash map? This way both a subset as well as all_ids can be fetched.